File: mail-examples/src/main/rb/io/vertx/example/mail/mail_eb.rb
require 'vertx-mail/mail_service'
mailService = VertxMail::MailService.create_event_bus_proxy($vertx, "vertx.mail")

email = {
  'bounceAddress' => "bounce@example.com",
  'to' => "user@example.com",
  'subject' => "this message has no content at all"
}

mailService.send_mail(email) { |result,result_err|
  if (result_err == nil)
    puts result
  else
    puts "got exception"
    result_err.print_stack_trace()
  end
}

File: mail-examples/src/main/js/io/vertx/example/mail/maileb.js
var MailService = require("vertx-mail-js/mail_service");
var mailService = MailService.createEventBusProxy(vertx, "vertx.mail");

var email = {
  "bounceAddress" : "bounce@example.com",
  "to" : "user@example.com",
  "subject" : "this message has no content at all"
};

mailService.sendMail(email, function (result, result_err) {
  if (result_err == null) {
    console.log(result);
  } else {
    console.log("got exception");
    result_err.printStackTrace();
  }
});
